external/fv3fit/tests/training/test_train.py

```diff
@@ -206,6 +227,9 @@ def test_train_default_model_on_identity(model_type, regtest):
     The model with default configuration options can learn the identity function,
     using gaussian-sampled data around 0 with unit variance.
     """
+    if model_type == "transformed":
+        pytest.skip()
```

Comment:
Do you know why your code doesn't pass this?

Reply:
No, but I think it's an intermittent error. To be solved another day I think. This PR added many tests so I think is still a step in the right direction.

Reply:
I'd like to pull down your PR and dig into it a little bit, if that's OK. Alternatively, have you already or can you test the refactored code on a production case?